src/files-and-uploads/data/utils.js

```diff
@@ -1,11 +1,54 @@
 import { InsertDriveFile, Terminal, AudioFile } from '@edx/paragon/icons';
 import { ensureConfig, getConfig } from '@edx/frontend-platform';
+import JSZip from 'jszip';
+import saveAs from 'file-saver';
 import FILES_AND_UPLOAD_TYPE_FILTERS from './constant';
 
 ensureConfig([
   'STUDIO_BASE_URL',
 ], 'Course Apps API service');
 
+export const getDownloadZipFolder = async (selectedRows, courseId, onError) => {
+  if (selectedRows.length > 1) {
+    const zip = new JSZip();
+    const date = new Date().toString();
+    const folder = zip.folder(`${courseId}-assets-${date}`);
+    const assetNames = [];
+    const assetFetcher = await Promise.all(
```

Comment:
If any one of the calls does not succeed then this promise gets rejected altogether. Is this the intention? Otherwise, Promise.allSettled allows more control over this.